test/common/http/conn_manager_impl_test_2.cc

```diff
@@ -4210,5 +4210,63 @@ TEST_F(HttpConnectionManagerImplTest, DownstreamTimingsRecordWhenRequestHeaderPr
   filter_callbacks_.connection_.raiseEvent(Network::ConnectionEvent::RemoteClose);
 }
 
+TEST_F(HttpConnectionManagerImplTest, PassMatchUpstreamSchemeHintToStreamInfo) {
```

Comment:
High-level comment:
it seems that the relevant part of the test is essentially:
`EXPECT_TRUE(filter->callbacks_->streamInfo().shouldSchemeMatchUpstream());`

Can this test be refactored to minimize the non-relevant parts?

Reply:
Cut as much of the test as I'm familiar with. Will need to dive deeper to see if any other cuts can be made.